app/controllers/users_controller.rb

```diff
@@ -45,6 +45,14 @@ def destroy
 
   private
 
+  def handle_update(user_params)
+    if user_params[:password].blank?
+      @user.update_without_password(user_params.slice(:email, :role))
```

Comment:
`update_without_password` [already deletes the password attributes](https://github.com/plataformatec/devise/blob/71fc5b351a8e8473733b0ca29d60c06e43a0efe1/lib/devise/models/database_authenticatable.rb#L98-L100), so you can just pass all `user_params`